opm/simulators/linalg/bda/Reorder.hpp

```diff
@@ -0,0 +1,120 @@
+/// \param[inout] rMat       reordered Matrix 
+template <unsigned int block_size>
+void blocked_reorder_matrix_by_pattern(BlockedMatrix *mat, int *toOrder, int *fromOrder, BlockedMatrix *rMat);
+
+/// Compute reorder mapping from the color that each node has received
+/// The toOrder, fromOrder and iters arrays must be allocated already
+/// \param[in] Nb            number of blocks in the vector
+/// \param[in] colors        array containing the number of the color that each row is assigned to
+/// \param[inout] toOrder    reorder pattern that lists for each index in the original order, to which index in the new order it should be moved
+/// \param[inout] fromOrder  reorder pattern that lists for each index in the new order, from which index in the original order it was moved
+/// \param[inout] iters      array containing for each color the number of rows that it contains
+void colorsToReordering(int Nb, std::vector<int>& colors, int *toOrder, int *fromOrder, int *iters);
+
+/// Reorder a vector according to the mapping in toOrder and fromOrder
+/// The rVector array must be allocated already
+/// \param[in] Nb            number of blocks in the vector
+/// \param[in] vector        vector to be reordered
+/// \param[in] toOrder       reorder pattern that lists for each index in the original order, to which index in the new order it should be moved
+/// \param[in] fromOrder     reorder pattern that lists for each index in the new order, from which index in the original order it was moved
+/// \param[inout] rVector    reordered vector
+template <unsigned int block_size>
+void blocked_reorder_vector_by_pattern(int Nb, double *vector, int *fromOrder, double *rVector);
+
+/// Determine whether all rows that a certain row depends on are done already
+/// \param[in] rowIndex      index of the row that needs to be checked for
+/// \param[in] rowPointers   row pointers of the matrix that the row is in
+/// \param[in] colIndices    column indices of the matrix that the row is in
+/// \param[in] doneRows      array that for each row lists whether it is done or not
+/// \return                  true iff all dependencies are done and if the result itself was not done yet
+bool canBeStarted(int rowIndex, int *rowPointers, int *colIndices, std::vector<bool>& doneRows);
+
+/// Find a level scheduling reordering for an input matrix
+/// The toOrder and fromOrder arrays must be allocated already
+/// \param[in] CSRColIndices  column indices array, obtained from storing the input matrix in the CSR format
+/// \param[in] CSRRowPointers row pointers array, obtained from storing the input matrix in the CSR format
+/// \param[in] CSCColIndices  row indices array, obtained from storing the input matrix in the CSC format
+/// \param[in] CSCRowPointers column pointers array, obtained from storing the input matrix in the CSC format
```

Comment:
Please compare explanations and variable name. They should correlate. As is this is a bit confusing.